team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -195,8 +193,11 @@ public TreeNode expand() {
 				children.add(new TreeNode(childState, childAction, this));
 			}
 
-			// Return the first child expanded
-			return children.get(0);
+			// Return a random child expanded - If we choose only the first node, it will just go down one path.
+			// This is part of why it was just moving to the sad corner
+			int index = (int) (Math.random() * children.size());
```

Comment:
I don't see how this affects our algorithm. This will only make a difference if the AI is unable to check all the nodes at the current depth - and that change will just mean picking randomly instead of only the first choice at the deepest part of the search. The only way this could possibly have an impact, is if we were to discover a heuristic which stated that in general, moves in certain regions of the board are more ideal than others - in which case, I would argue we should bias towards those locations rather than using randomness anyway.

That all said, I don't mind the change. While it's effectively meaningless, having the AI pick randomly will make it feel more like a real decision if the AI does not manage to search the tree to a meaningful depth.